cli/command/engine/init.go

```diff
@@ -0,0 +1,64 @@
+package engine
+
+import (
+	"context"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/containerizedengine"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	defaultEngineVersion = cli.Version
+)
+
+type initOptions struct {
+	version        string
+	image          string
+	registryPrefix string
+	configFile     string
+}
+
+func newInitCommand(dockerCli command.Cli) *cobra.Command {
+	options := initOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "init [OPTIONS]",
+		Short: "Initialize a local engine",
+		Long: `This command will initialize a local engine running on containerd.
+
+Configuration of the engine is managed through the daemon.json configuration
+file on the host and may be pre-created before running the 'init' command.
+`,
+		Args: cli.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runInit(dockerCli, options)
+		},
+		Annotations: map[string]string{"experimentalCLI": ""},
+	}
+	flags := cmd.Flags()
+	flags.StringVar(&options.version, "version", defaultEngineVersion, "Specify engine version")
+	flags.StringVar(&options.image, "engine-image", containerizedengine.CommunityEngineImage, "Specify engine image")
+	flags.StringVar(&options.registryPrefix, "registry-prefix", "docker.io/docker", "Override the default location where engine images are pulled")
+	flags.StringVar(&options.configFile, "config-file", "/etc/docker/daemon.json", "Specify the location of the daemon configuration file on the host")
```

Comment:
Hmmm what about windows 🤔 ? Should this feature work with containerd on windows? If so should the default path be specified depending the platform? WDYT @thaJeztah @vdemeester ?

Reply:
We're working towards supporting windows, but we're going to keep the `init` CLI UX experimental for now in the upcoming release since we don't quite have time to get all the details sorted out in time for FC.

That said, definitely open to suggestions on refinements to get this better aligned for windows support.